src/QuadForm/indefiniteLLL.jl

```diff
@@ -0,0 +1,357 @@
+
+################################################################################
+#
+#                       indefinite LLL reduction
+#
+################################################################################
+
+################################################################################
+#                  Helpful
+################################################################################
+
+#=
+  _round_matrix(M::MatElem) -> MatElem
+
+Returns the matrix $M$ where all entries are replaced by their rounded values.
+
+$Example$
+==========
+julia> _round_matrix(QQ[1//3 -2 3//2; 5//3 0 -6; -7//3 -81//50 0])
+[0 -2 2]
+[2 0 -6] 
+[-2 -2 0]
+=#
+function _round_matrix(M::MatElem)
+  return map_entries(round, M)
+end
+
+
+#=
+  _abs_matrix(M::MatElem) -> MatElem
+
+Returns the matrix $M$ where all entries are replaced by their absolute values.
+
+$Example$
+==========
+julia> _abs_matrix(ZZ[1 -2 3; 4 0 -6; -7 -81 0])
+[1 2 3]
+[4 0 6] 
+[7 81 0]
+=#
+function _abs_matrix(M::MatElem)
+    return map_entries(abs, M)
+end
+
+#=
+  vecextract(M::MatElem,x::Int64,y::Int64) -> MatElem
+
+Extracts components of $M$ with regards to the binary representation of $x$ and $y$. 
+The location of the one entries give the position of the entries which should be extracted.
+For example if $x$ = $y$ = 13 = (1101)_2 then M[[1,3,4], [1,3,4]] is returned.
+
+$Example$
+==========
+julia> vecextract(ZZ[1 2 3; 4 5 6; 7 8 9],6,3)
+[4 5]
+[7 8]
+=#
+function _vecextract(M::MatElem,x::Int64,y::Int64)
+  x_bin = digits(x,base = 2)
+  y_bin = digits(y, base = 2)
+
+  list_x = [i for i = 1:length(x_bin) if x_bin[i] == 1]
+  list_y = [i for i = 1:length(y_bin) if y_bin[i] == 1]
+
+  return M[list_x , list_y]
+
+end
+
+function _vecextract(M::MatElem, x::Union{Int64,Vector{Int64}},y::Int64)
+  y_bin = digits(y, base = 2)
+  list_y = [i for i = 1:length(y_bin) if y_bin[i] == 1]
+
+  return M[x , list_y]
+
+end
+
+################################################################################
+#                           linear algebra
+################################################################################
+
+#=
+    _mathnf(A::MatElem) -> MatElem, MatElem
+
+Given a rectangular matrix $A$ of dimension $nxm$. Computes the Hessian matrix $H$ of dimension $nxm$ 
+and the unimodular transformation matrix $U$ such that $AU$ = $H$.
+=#
+function _mathnf(A::MatElem)
+  H, U = hnf_with_transform(reverse_cols(transpose(A)))
+  H = reverse_rows(reverse_cols(transpose(H)))
+  U = reverse_cols(transpose(U))
+
+  return H, U
+end
+
+#=
+    _complete_to_basis(v::MatElem; redflag::Bool = false) -> MatElem
+
+Given a rectangular matrix $nxm$ with $n$ != $m$ and redflag = 0.
+Computes a unimodular matrix with the last column equal to the last column of $v$.
+If redflag = 1, it LLL-reduce the $n$-$m$ first columns if $n$ > $m$.
+=#
```

Comment:
The documentation is a bit confused: first, you should not put a " . " at the end of the first line because it is not a sentence, it should a be coma and linked to the second line. Then you can't fixed a variable argument in the documentation so the `redflag = 0` should not appear there. I have a problem with the consistency of the output: if `n = m`, then the return the input matrix, otherwise an unitary (not unimodular) transformation... I don't know the precise use, but I guess it would be more appropriate to return the identity matrix if `n = m` and adapt the rest of the code there of. You have to avoid different output types/kind of output depending on the input. 
Don't forget that you are using boolean, so it should be " If `redflag == false`, it performs an LLL-reduction of the first $n-m$ columns". Finally, I am concerned by the fact that if the matrix entry has more columns than rows, then your algorithm won't work. There should be something taking care of this part, or an input chec forcing that $n \geq m$.